modules/globebrowsing/tile/rawtiledatareader/rawtiledatareader.cpp

```diff
@@ -0,0 +1,416 @@
+        preprocessData->hasMissingData[c] = false;
+        noDataValues[c] = noDataValueAsFloat();
+    }
+
+    for (size_t y = 0; y < region.numPixels.y; y++) {
+        size_t yi = (region.numPixels.y - 1 - y) * bytesPerLine;
+        size_t i = 0;
+        for (size_t x = 0; x < region.numPixels.x; x++) {
+            for (size_t c = 0; c < _dataLayout.numRasters; c++) {
+                float noDataValue = noDataValueAsFloat();
+                float val = tiledatatype::interpretFloat(
+                    _dataLayout.glType,
+                    &(rawTile->imageData[yi + i])
+                );
+                if (val != noDataValue) {
+                    preprocessData->maxValues[c] = std::max(
+                        val,
+                        preprocessData->maxValues[c]
+                    );
+                    preprocessData->minValues[c] = std::min(
+                        val,
+                        preprocessData->minValues[c]
+                    );
+                }
+                else {
+                    preprocessData->hasMissingData[c] = true;
+                }
+                i += _dataLayout.bytesPerDatum;
+            }
+        }
+    }
+
+    for (size_t c = 0; c < _dataLayout.numRasters; c++) {
+        if (preprocessData->maxValues[c] > 8800.0f) {
+            //LDEBUG("Bad preprocess data: " << preprocessData->maxValues[c] << " at " << region.tileIndex);
+        }
+    }
+
+    return std::shared_ptr<TileMetaData>(preprocessData);
+}
+
+float RawTileDataReader::depthOffset() const {
+    return 0.0f;
+}
+
+float RawTileDataReader::depthScale() const {
+    return 1.0f;
+}
+
+TileDepthTransform RawTileDataReader::calculateTileDepthTransform() {
+    bool isFloat =
+        (_dataLayout.glType == GL_FLOAT || _dataLayout.glType == GL_DOUBLE);
+    double maximumValue =
+        isFloat ? 1.0 : tiledatatype::getMaximumValue(_dataLayout.glType);
+
+    TileDepthTransform transform;
+    transform.depthOffset = depthOffset();
+    transform.depthScale = depthScale() * maximumValue;
+    return transform;
+}
+
+RawTile::ReadError RawTileDataReader::postProcessErrorCheck(
+    std::shared_ptr<const RawTile> rawTile, const IODescription& io)
+{
+    ensureInitialized();
+
+    double missingDataValue = noDataValueAsFloat();
+
+    bool hasMissingData = false;
+        
+    for (size_t c = 0; c < _dataLayout.numRasters; c++) {
+        hasMissingData |= rawTile->tileMetaData->maxValues[c] == missingDataValue;
+    }
+        
+    bool onHighLevel = rawTile->tileIndex.level > 6;
+    if (hasMissingData && onHighLevel) {
+        return RawTile::ReadError::Fatal;
+    }
+    // ugly test for heightmap overlay
+    if (_dataLayout.textureFormat.ghoulFormat == ghoul::opengl::Texture::Format::RG) {
+        // check the alpha
+        if (rawTile->tileMetaData->maxValues[1] == 0.0
+            && rawTile->tileMetaData->minValues[1] == 0.0)
+        {
+            //return CE_Warning;
```

Comment:
Not exactly sure what that was supposed to do but we do not have heightmap overlays anymore so I removed that